generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/FixedField.java

```diff
@@ -44,8 +44,8 @@
             });
     }
 
-    public FieldSpec getFieldSpecForValues(){
-        return this.valuesFieldSpec;
+    boolean hasValueSet() {
+        return current == null || !current.equals(NOT_ITERATED);
```

Comment:
This should be `== NOT_ITERATED`, you DO want to use reference equals explicitly rather than by accident.